### ISSUE TITLE
client.close callback never invoked

### ISSUE THREAD

**Issue by whitelizard** [deepstreamIO/deepstream.io-client-js#458]
It seems that the callback, documented to be called when connection is closed with `client.close`, is never invoked.

It also seems that it is not tested for in the tests. ([Here](https://github.com/deepstreamIO/deepstream.io-client-js/blob/v4/test/connection/connectionSpec.ts))

Replicate the bug (with a plain DS server running in parallel, and running the following in babel-node):
```
> var ds = require('deepstream.io-client-js');
> var c = ds('localhost:6020');
> c.login({});
> c.close(state => { console.log('CLOSED!', state) });
```

**Comment by Thomas-P** [comment]
Hi, I added a fix for 2.3.0, #461

**Comment by yasserf** [comment]
This was merged a while back. Thank you!
